internal/pkg/config/filterType.go

```diff
@@ -0,0 +1,56 @@
+package config
+
+import "fmt"
+
+type FilterType int
+
+const (
+	loadBalancer FilterType = iota
+	securityGroup
+)
+
+var LoadBalancerStr = "loadBalancer"
+var SecurityGroupStr = "securityGroup"
+
+func (filterType FilterType) String() string {
+	filterTypes := [...]string{
+		"loadBalancer",
+		"securityGroup",
+	}
+
+	if filterType < loadBalancer || filterType > securityGroup {
+		return "Unknown"
+	}
+
+	return filterTypes[filterType]
+}
+
+func toFilterType(s string) (u FilterType, err error) {
+	switch s {
+	case "loadBalancer":
+		u = loadBalancer
+
+	case "securityGroup":
+		u = securityGroup
+
+	default:
+		err = fmt.Errorf("incorrect FilterType :%s provided", s)
+	}
+	return
+}
+
+func (filterType *FilterType) UnmarshalYAML(unmarshal func(interface{}) error) error {
```

Comment:
Does this method automatically get called when unmarshalling a yaml?

Reply:
yes it does